File: algo/sort/js/selection.js
const selection = (nums) => {
    for(let i = 0; i < nums.length; i++) {

        let select = i;
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[j] < nums[select]) {
                select = j;
            }
        }

        if (select != i) {
            [ nums[select], nums[i] ] = [ nums[i], nums[select] ];
        }
    }
}

// test here
let nums = [5, 4, 3, 2, 1, 6, 0];
selection(nums);
console.log(nums);

File: algo/sort/java/Selection.java
import java.util.Arrays;

public class Selection {
    public static void main(String[] args){
        int[] arr = { 8, 5, 4, 2, 0, -2, 4, 8, 0 };
        Selection.selection(arr);
        System.out.println(Arrays.toString(arr));
    }

    public static void selection(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            int select = i;
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[j] < nums[select]) {
                    select = j;
                }
            }

            if (select != i) {
                int temp = nums[select];
                nums[select] = nums[i];
                nums[i] = temp;
            }
        }
    }
}
